pkg/flow/events.go

```diff
@@ -1005,3 +1015,175 @@ func (events *events) listenForEvents(ctx context.Context, im *instanceMemory, c
 	return nil
 
 }
+
+func (flow *flow) ApplyCloudEventFilter(ctx context.Context, in *grpc.ApplyCloudEventFilterRequest) (*grpc.ApplyCloudEventFilterResponse, error) {
+
+	flow.sugar.Debugf("Handling gRPC request: %s", this())
+
+	resp := new(grpc.ApplyCloudEventFilterResponse)
+
+	namespace := in.GetNamespace()
+	filtername := in.GetFilterName()
+	cloudevent := in.GetCloudevent()
+
+	var script string
+
+	var key keyPair
+	key.filtername = filtername
+	key.namespace = namespace
+
+	ns, err := flow.getNamespace(ctx, flow.db.Namespace, namespace)
+	if err != nil {
+		return resp, err
+	}
+
+	if jsCode, ok := cache[key]; ok {
+		script = jsCode
+
+	} else {
+		ceventfilter, err := ns.QueryCloudeventfilters().Where(enteventsfilter.NameEQ(filtername)).Only(ctx)
+		if err != nil {
+			return resp, err
+		}
+
+		script = ceventfilter.Jscode
+
+	}
+
+	var mapEvent map[string]interface{}
+	err = json.Unmarshal(cloudevent, &mapEvent)
+
+	if err != nil {
+		return resp, err
+	}
+
+	//create js runtime
+	vm := goja.New()
+
+	vm.Set("event", mapEvent)
+
+	_, err = vm.RunString(script)
+	if err != nil {
+		//panic(err)
+		return resp, err
+	}
+
+	var fn func() map[string]interface{}
+	err = vm.ExportTo(vm.Get("filter"), &fn)
+	if err != nil {
+		//panic(err)
+		return resp, err
+	}
+
+	newEventMap := fn()
+	newBytesEvent, err := json.Marshal(newEventMap)
+	if err != nil {
+		return resp, err
+	}
+
+	resp.Event = newBytesEvent
+
+	if string(resp.GetEvent()) == "null" {
+		event, _ := EventByteToCloudevent(cloudevent)
```

Comment:
Assigning the error to `_` means the error is ignored and the subsequent check is faulty.